### PR TITLE
Create an examples folder and add a minimal example

### PR DESCRIPTION
**Pull request by alxclark**
## Description

Added an example for showcasing a working example and make it easier for us to debug issues.

Heavily inspired by https://github.com/robin-drexler/checkout-ui-usage-example, I've mostly copied over the structure and remove the yarn workspaces and custom package.

## Notes

* I'll need to fix linting to avoid to check the examples folder

### CODE REVIEW

File: examples/minimal/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  "main": "index.js",
+  "license": "MIT",
+  "private": "true",
+  "scripts": {
+    "start": "NODE_ENV=development webpack-dev-server --config ./webpack.config.js",
+    "build": "NODE_ENV=production webpack --config ./webpack.config.js",
+    "serve": "cd dist && http-serve"
+  },
+  "dependencies": {
+    "@shopify/checkout-ui-react": "latest",
```
Comment (alxclark) [inline examples/minimal/package.json:13]: This uses the latest version of the package, to me this is acceptable and doesn't force us to keep examples up to date when we are in between two versions. We could also simply pin the version.

File: examples/minimal/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# `@shopify/checkout-ui-react` minimal example
+
+```sh
+yarn
+```
+
+## Run dev server
```
Comment (robin-drexler) [inline examples/minimal/README.md:7]: I know it's logged at the beginning of `yarn start`, but maybe we should add where the app is served to the readme? (http://localhost:8080/)

File: examples/minimal/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+
+const isProduction = process.env.NODE_ENV === 'production';
+
+module.exports = addWebpackConfig(
+  {
+    entry: './src/index.js',
+    resolve: {
+      alias: {
+        '@shopify/argo-checkout-react': '@shopify/checkout-ui-react',
```
Comment (lemonmade) [inline examples/minimal/webpack.config.js:14]: The addWebpackConfig function doesn't do this?

File: examples/minimal/package.json (hunk 1 above)
Comment (lemonmade) [reply examples/minimal/package.json:13]: I don't even know the behavior of latest with a lockfile, but I would pin the version anyways, too.

File: examples/minimal/package.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "checkout-ui-minimal",
+  "version": "1.0.0",
+  "main": "index.js",
+  "license": "MIT",
+  "private": "true",
```
Comment (lemonmade) [inline examples/minimal/package.json:6]: I think this is meant to be a boolean

**Review by lemonmade — APPROVED** [review]

**Pull request closed without merging**